src/mats_l1_processing/photometer.py

```diff
@@ -0,0 +1,93 @@
+
+def calibrate_pm(df: DataFrame, photometer: Photometer):
+    
+# ===============================================================================================================
+# Calibration starts here
+# PM1 is Bkg photometer, 754BP3 nm / PM2 is A-band photometer, 763BP8 nm
+#   Change to bits/s
+    PM1_Sig_bit = df['PM1S']/df['PM1SCNTR']
+    PM1_Tpd_bit = df['PM1A']/df['PM1ACNTR']
+    PM1_Tif_bit = df['PM1B']/df['PM1BCNTR']
+    PM2_Sig_bit = df['PM2S']/df['PM2SCNTR']
+    PM2_Tpd_bit = df['PM2A']/df['PM2ACNTR']
+    PM2_Tif_bit = df['PM2B']/df['PM2BCNTR']
+    pm_texp = round(df['PM1SCNTR']/166) # exposure time in seconds, 1 second "integration" is 166-167 sampling cycles
+
+# ===========================================================
+# Calibrate data
+
+#   extracting variables from dict
+    bitar = photometer.cal_therm['bitar'] # 0.1 to 4095 bits in steps of 0.1 bit, array of float64 (1,40950)
+    TempFM1if_raw = photometer.cal_therm['TempFM1if_raw'] # array of float64 (1,40950)
+    TempFM1pd_raw = photometer.cal_therm['TempFM1pd_raw'] # array of float64 (1,40950)
+    TempFM2if_raw = photometer.cal_therm['TempFM2if_raw'] # array of float64 (1,40950)
+    TempFM2pd_raw = photometer.cal_therm['TempFM2pd_raw'] # array of float64 (1,40950)
+    Temperatur = photometer.cal_rad['Temperatur'] # -20 to +44 °C in steps of 0.1 °C,  # array of float64 (1,641)
+    SignFM1_Rad_raw = photometer.cal_rad['SignFM1_Rad_raw'] # 2D matrix, array of float64 (641,40950)
+    SignFM2_Rad_raw = photometer.cal_rad['SignFM2_Rad_raw'] # 2D matrix, array of float64 (641,40950)
+
+# =================================
+# Change raw temperature data to °C
+#   define pmBkg_Tpd, pmBkg_Tif, pmAband_Tpd, pmAband_Tif
+    pmBkg_Tpd = np.NaN * np.ones_like(PM1_Tpd_bit)   # Temperature of the Bkg photometer photodiode
+    pmBkg_Tif = np.NaN * np.ones_like(PM1_Tif_bit)   # Temperature of the Bkg photometer filter
+    pmAband_Tpd = np.NaN * np.ones_like(PM2_Tpd_bit) # Temperature of the A-band photometer photodiode
+    pmAband_Tif = np.NaN * np.ones_like(PM2_Tif_bit) # Temperature of the A-band photometer filter
+
+    for ij in range(0, len(PM1_Tpd_bit)-1):
+        index11 =  np.where(bitar == round(PM1_Tpd_bit[ij],1))
+        pmBkg_Tpd[ij] = TempFM1pd_raw[index11]
+        index12 =  np.where(bitar == round(PM1_Tif_bit[ij],1))
+        pmBkg_Tif[ij] = TempFM1if_raw[index12]
+        index13 =  np.where(bitar == round(PM2_Tpd_bit[ij],1))
+        pmAband_Tpd[ij] = TempFM2pd_raw[index13]
+        index14 =  np.where(bitar == round(PM2_Tif_bit[ij],1))
+        pmAband_Tif[ij] = TempFM2if_raw[index14]
+
+# ==============================
+# Change raw photometer data to photons cm-2 str-1 s-1
+#   define pmBkg_Sig, pmAband_Sig
+    pmBkg_Sig = np.ones_like(PM1_Sig_bit)   # Background photometer signal
+    pmAband_Sig = np.ones_like(PM2_Sig_bit) # A-band photometer signal
+    #ijk1=0
+    #ijk2=0
+
+    for ik in range(0, len(PM1_Sig_bit)-1):
```

Comment:
Why the `-1`? Last index of no interest? (Also, the initial `0` is redundant.)

``` python
In [1]: list(range(0, 5))
Out[1]: [0, 1, 2, 3, 4]

In [2]: list(range(5))
Out[2]: [0, 1, 2, 3, 4]
```

Reply:
no idea can you answer this @jonashedin

Reply:
:-) Likely my bad python skills, I can not recall any reason to leave out the last index.

Reply:
```suggestion
    for ik in range(len(PM1_Sig_bit)):
```

Reply:
fixed